src/journal/journal-file.c

```diff
@@ -3247,10 +3250,21 @@ int journal_file_open(
 #elif HAVE_XZ
         f->compress_xz = compress;
 #endif
+
+        if (compress_threshold_bytes == (uint64_t) -1)
+                f->compress_threshold_bytes = DEFAULT_COMPRESS_THRESHOLD;
+        else
+                f->compress_threshold_bytes = MAX(MIN_COMPRESS_THRESHOLD, compress_threshold_bytes);
```

Comment:
I think when `compress_threshold_bytes < MIN_COMPRESS_THRESHOLD` we should return `-EINVAL` or at least warn about that, instead of silently ignoring the argument and using `MIN_COMPRESS_THRESHOLD`.

Reply:
The reason I'm torn here is that we may want to set a different minimum depending upon the algorithm. For example, the effective minimum for XZ is 80 bytes, but LZ4 can do well with less.

By punting, we don't accidentally build the limit into the ABI and potentially break folks later. I think that's okay, because this is an optimization thing.

Reply:
OK.